src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
```

Comment:
Since TypeScript automatically creates class properties from constructor parameters when using an access modifier, you don't need to manually assign them. You can remove the assignments and leave the constructor parameters as they are.